### PR TITLE
read and write updates

### PR DESCRIPTION
**Pull request by nicholas-masel**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/read_dataset_json.R [hunk 1]
```diff
@@ -48,59 +48,49 @@ read_dataset_json <- function(file) {
...
+  date_cols <- tt %in% c("date") & tdt %in% "integer"
+  datetime_cols <- tt %in% c("datetime", "time") & tdt %in% "integer"
+  d[date_cols] <- lapply(d[date_cols], as.Date)
+  d[datetime_cols] <- lapply(d[datetime_cols], as.POSIXct)
 
-  # Apply variable labels
+  # Apply variable attributes
   d[names(d)] <- lapply(items$name, set_col_attr, d, 'label', items)
-  d[names(d)] <- lapply(items$name, set_col_attr, d, 'OID', items)
+  d[names(d)] <- lapply(items$name, set_col_attr, d, 'itemOID', items)
```
Comment (mstackhouse) [inline R/read_dataset_json.R:75]:
```suggestion
  d[names(d)] <- lapply(items$name, set_col_attr, d, 'itemGroupOID', items)
```

Comment (nicholas-masel) [reply R/read_dataset_json.R]: This was moved to the columns attribute on the data metadata.

**Comment by nicholas-masel** [comment]
@mstackhouse I added variable metadata argument to write, since this is not on the datasetjson object anymore.  I can change the write function if we want to move it back to datasetjson.

File: R/write_dataset_json.R [hunk 2]
```diff
@@ -10,7 +10,7 @@
 #' @examples
 #' # Write to character object
 #' ds_json <- dataset_json(iris, "IG.IRIS", "IRIS", "Iris", iris_items)
-#' js <- write_dataset_json(ds_json)
+#' js <- write_dataset_json(ds_json, iris_items)
```
Comment (mstackhouse) [inline R/write_dataset_json.R:13]: I think we can remove this again

File: tests/testthat/test-validate_dataset_json.R [hunk 3]
```diff
@@ -1,19 +1,33 @@
...
-  expect_message(validate_dataset_json(js), "File is valid per the Dataset JSON v1.0.0 schema")
+  expect_message(validate_dataset_json(js), "File is valid per the Dataset JSON v1.1.0 schema")
 
 })
 
 test_that("JSON can checked from URL", {
-  fpath <- paste0("file://", normalizePath(test_path("testdata", "ae.json")))
+  fpath <- paste0("file://", normalizePath(test_path("testdata", "invalid_dm.json")))
   expect_warning(
     err <- validate_dataset_json(fpath),
```
Comment (mstackhouse) [inline tests/testthat/test-validate_dataset_json.R:27]: Do you think this is valuable and worth the dependency?

**Review by mstackhouse — APPROVED** [review]
